src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala

```diff
@@ -573,29 +580,71 @@ class ResourceServiceSpec extends FlatSpec with Matchers with TestSupport with B
+    val policy = (AccessPolicy.id composeLens ResourceAndPolicyName.resource composeLens Resource.resourceTypeName).set(constrainableResourceType.name)(genPolicy.sample.get)
+    val resource = policy.id.resource
+    val viewPolicyName = AccessPolicyName(constrainableReaderRoleName.value)
+
+    val res = for{
+      _ <- constrainableService.createResourceType(constrainableResourceType).unsafeToFuture()
+      _ <- constrainableService.createResourceType(managedGroupResourceType).unsafeToFuture()  // make sure managed groups in auth domain set are created. dummyUserInfo will be member of the created resourceId
+      _ <- Future.traverse(resource.authDomain)(a => managedGroupService.createManagedGroup(ResourceId(a.value), dummyUserInfo))
+      // create resource that dummyUserInfo is a member of for constrainableResourceType
+      _ <- constrainableService.createResource(constrainableResourceType, resource.resourceId, Map(viewPolicyName -> constrainablePolicyMembership), resource.authDomain, dummyUserInfo)
+      r <- constrainableService.listUserAccessPolicies(constrainableResourceType.name, dummyUserInfo.userId).unsafeToFuture()
+    } yield r
+
+    val expected = Set(UserPolicyResponse(resource.resourceId, viewPolicyName, resource.authDomain, Set.empty))
+    res.futureValue(Timeout(Span(10, Seconds))) shouldBe expected
+  }
+
+  it should "list required authDomains and authDomains user is not a member of if constrainable" in {
+    val user = genUserInfo.sample.get
+    val policy = (AccessPolicy.id composeLens ResourceAndPolicyName.resource composeLens Resource.resourceTypeName).set(constrainableResourceType.name)(genPolicy.sample.get)
+    val resource = policy.id.resource
+    val viewPolicyName = AccessPolicyName(constrainableReaderRoleName.value)
+
+    val res = for{
+      _ <- constrainableService.createResourceType(constrainableResourceType).unsafeToFuture()
+      _ <- constrainableService.createResourceType(managedGroupResourceType).unsafeToFuture()
+      _ <- Future.traverse(resource.authDomain)(a => managedGroupService.createManagedGroup(ResourceId(a.value), dummyUserInfo))
+      // create resource that dummyUserInfo is a member of for constrainableResourceType
+      _ <- constrainableService.createResource(constrainableResourceType, resource.resourceId, Map(viewPolicyName -> constrainablePolicyMembership), resource.authDomain, dummyUserInfo)
+      _ <- dirDAO.createUser(WorkbenchUser(user.userId, Some(GoogleSubjectId(user.userId.value)), user.userEmail))
+      _ <- constrainableService.createPolicy(policy.id, policy.members + user.userId, policy.roles, policy.actions)
+      r <- constrainableService.listUserAccessPolicies(constrainableResourceType.name, user.userId).unsafeToFuture()
+    } yield r
+
+    val expected = Set(UserPolicyResponse(resource.resourceId, policy.id.accessPolicyName, resource.authDomain, resource.authDomain))
+    res.futureValue(Timeout(Span(10, Seconds))) shouldBe expected
+  }
+
```

Comment:
add a test for listUserAccessPolicies where there is a resource in a constrainable type but does not have any auth domains